apps/src/templates/studioHomepages/TeacherHomepage.jsx

```diff
@@ -47,12 +51,21 @@ export class UnconnectedTeacherHomepage extends Component {
     teacherEmail: PropTypes.string,
     schoolYear: PropTypes.number,
     specialAnnouncement: shapes.specialAnnouncement,
-    beginGoogleImportRosterFlow: PropTypes.func
+    beginGoogleImportRosterFlow: PropTypes.func,
+    schoolHasDonor: PropTypes.bool
   };
 
+  constructor(props) {
+    super(props);
+    this.hideTeacherOfTheYearBanner = this.hideTeacherOfTheYearBanner.bind(
```

Comment:
I always get confused about when to bind vs. not, can you explain this usage to me for my understanding?

Reply:
Honestly I've watched videos about `this` and binding so many times and it still confuses me a bit! But generally if I'm referencing `this` in the function then I'll bind it to make sure when I call `this.setState` that it sets the state of the right component. (I think that if you're only calling the function from within the component then you don't need to bind, but in this case I'm passing the function down to a child component so I want to make sure it's bound)

Reply:
haha glad I'm not alone, that makes sense thanks for the explanation!